### DISCUSSION TITLE
Differences Between v4 and v5

### DISCUSSION THREAD

**Discussion by JKomoroski** [dropwizard/dropwizard#8103] [General]
Version 3 and 4 were announced with some discussion and context around what the breaking changes were and what the upgrade paths would look like:
https://github.com/dropwizard/dropwizard/discussions/4720

I was unable to find a similar discussion about version 4 to version 5. Is there a summary of breaking changes that require a move to version 5 yet?

**Comment by joschi** [comment]
> I was unable to find a similar discussion about version 4 to version 5. Is there a summary of breaking changes that require a move to version 5 yet?

The main change from Dropwizard 4.x to 5.x (as of now) will be the move to Jakarta EE 10 with all dependencies such as Eclipse Jetty, and using Java 17 as baseline.

**Comment by JKomoroski** [comment]
> using Java 17 as baseline.

That's huge. Also Exciting. 

Am I safe to assume I can watch the dropwizard.io upgrade notes page for finalized information about version 5 in the future? ie: [https://www.dropwizard.io/en/stable/manual/upgrade-notes/upgrade-notes-4_0_x.html]()

**Reply by joschi** [reply]
> That's huge. Also Exciting.

Please note that Dropwizard 3.x and 4.x (and even 2.1.x) also work with Java 21.

This is about the internal Java language baseline of the Dropwizard framework.

> Am I safe to assume I can watch the dropwizard.io upgrade notes page for finalized information about version 5 in the future?

Yes, we branched off the `release/5.0.x` branch barely a week ago, so give us some time. 😉

**Reply by sleberknight** [reply]
We've been using Dropwizard 4.x on Java 17 for a while now and it works great. Before updating to 4.x we were running DW 2.1.x on Java 17.

**Comment by JKomoroski** [comment]
Thanks for the context here. We are mostly on 11, with some 17. I'll make sure that the team is aware that 17 will be a hard requirement by the time v5 comes out. I'll close this discussion for now.
